src/app/shared/modules/sai-editor/sai-editor.service.ts

```diff
@@ -92,7 +92,7 @@ export class SaiEditorService extends MultiRowComplexKeyEditorService<SmartScrip
   }
 
   async generateComments() {
-    for (const row of this._newRows) {
+    for (const row of structuredClone(this._newRows)) {
```

Comment:
it would be nice to have a unit test that would fail before this bug was fixed, if possible please add